### ISSUE TITLE
default transpilation generates incorrect result when using statevector simulation

### ISSUE THREAD

**Issue by chunfuchen** [Qiskit/qiskit-terra#2980]
### Information

- **Qiskit Terra version**: the master
- **Python version**: 3.7.3
- **Operating system**: macOS 10.14

### What is the current behavior?
the transpiler transpiles the circuits and the result of statevector simulation is incorrect but qasm simulator is okay.

Circuit before transpile
![image](https://user-images.githubusercontent.com/7079318/62982837-76422c80-bdfb-11e9-9640-546d39fc8a77.png)
After:
![image](https://user-images.githubusercontent.com/7079318/62982846-7d693a80-bdfb-11e9-9199-4815ced3c30f.png)

### Steps to reproduce the problem
```python
from qiskit import QuantumCircuit, QuantumRegister, BasicAer, execute
from qiskit.quantum_info import state_fidelity
from qiskit.compiler import transpile

backend = BasicAer.get_backend('statevector_simulator')

qr = QuantumRegister(2)
qc = QuantumCircuit(qr)

qc.y(qr[0])
qc.x(qr[1])
qc.cx(qr[0], qr[1])

t_qc = transpile(qc, coupling_map=[[0, 1]])

print(qc)
print(t_qc)

state_wo_t = execute(qc, backend=backend, optimization_level=0).result().get_statevector()
state_w_t = execute(t_qc, backend=backend).result().get_statevector()
print(state_fidelity(state_wo_t, state_w_t))
```

### What is the expected behavior?
the transpile should assure the quantum state is the same up to global phase (state fidelity should be close to 1.)

Furthermore, why the transpiled circuit generates h gates?

### Suggested solutions
I do not know, this issue happened after the DenseLayout is used for default transpiler.

**Comment by mtreinish** [comment]
The dense layout change is being reverted here: https://github.com/Qiskit/qiskit-terra/pull/2975 Does that fix the issue for you?

**Comment by nonhermitian** [comment]
How does the dense layout break this?  This is having the cx reversed for no reason.

**Comment by chunfuchen** [comment]
@mtreinish yes, after I pull the newest change, it works. 
But may I know that at which optimization level DenseLayout will be used?

**Comment by nonhermitian** [comment]
You get the same incorrect answer if you do:
```
transpile(qc, coupling_map=[[0, 1]], optimization_level=1, initial_layout=[1,0])
```
because the state vector simulator does not take into account the layout.

**Comment by chunfuchen** [comment]
Yes, the transpiler permutates the qubit mapping; however, the statevector always considers the first qubit is qubit 0, then the statevector result is incorrect. Is this users' responsible to post-process the statevector?

**Comment by nonhermitian** [comment]
It is not incorrect, it is not just in the same order as the other one. Just set initial layout to list(range(n_qubits)) and it will work at all optimization levels.
